### PR TITLE
translating admonitions

### PR DESCRIPTION
**Pull request by michalfapso**
Thanks a lot for working on this project!

I have a documentation with multiple languages and I was looking for a way to translate admonitions.

I wasn't sure how should I add a test for it, whether to add an admonition into `tests/docs_*_two_languages` or should I create a separate testing doc?

### CODE REVIEW

**Comment by ultrabug** [comment]
Thanks for this amazing work @michalfapso 

I'd rather you add tests to a separate tree so that it's strictly related to the feature and easier to cope with

Please rebase

**Comment by michalfapso** [comment]
Sure, now I've pushed a rebased commit with tests included.

**Comment by ultrabug** [comment]
@michalfapso I forgot to ask sorry but could you share a rationale of the problem you're trying to solve with an example as to why it's not possible currently?

**Comment by michalfapso** [comment]
Sure, @ultrabug, I have a documentation in English with quite a lot of admonitions. Then I have auto-translated it to several other languages which would be later manually corrected by a community. I'm using the material theme and your static-i18n plugin for it.

When I have `!!! tip` in my English documentation it gets rendered as "Tip" (as seen here: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types). To translate it to another language I can write `!!! tip "Conseil"` which replaces the admonition title with "Conseil", but I would prefer to have a translation map for each admonition type so that its translated across the whole documentation consistently.

I could have written a separate script just to post process all markdown files and to add the translation of admonition titles directly to them, but I prefer not having those translations hard coded in all markdowns.

I already have some strings translated via `nav_translations`, so I find it convenient to have the proposed `admonition_tranlstations` right beside it in the `mkdocs.yml`.

Here are some related requests:
https://github.com/squidfunk/mkdocs-material/discussions/4599
https://github.com/squidfunk/mkdocs-material/issues/749

File: mkdocs_static_i18n/plugin.py [hunk 1]
```diff
@@ -124,6 +125,27 @@ def on_template_context(self, context, template_name, config):
...
+        """
+        The page_markdown event is called after the page's markdown is loaded from file
+        and can be used to alter the Markdown source text.
+
+        Here we translate admonition types
+        """
+        admonition_translations = self.current_language_config.admonition_translations or {}
+        out = []
+        for line in markdown.splitlines():
+            m = re.match(r'^(!!!\s+)([^\s]+)(.*)$', line)
```
Comment (ultrabug) [inline mkdocs_static_i18n/plugin.py:139]: Defining admonitions as `!!!tip` is also valid and I think that this regex does not match it.

If I'm not mistaken, please add tests for this behavior too!

**Comment by ultrabug** [comment]
Makes sense and I'm honored that the plugin could help people find a solution to a 2018 problem :+1: thanks

Please check my comments

File: mkdocs_static_i18n/plugin.py (hunk 1 above)
Comment (michalfapso) [reply mkdocs_static_i18n/plugin.py:139]: That's a good point! I haven't seen it in documentation, but you're right. So I've copied the regex from here now:
https://github.com/Python-Markdown/markdown/blob/master/markdown/extensions/admonition.py#L53

File: docs/setup/translating-content.md [hunk 2]
```diff
@@ -30,3 +30,42 @@ site
 Which means that the `image.png` and its `fr/image.png` localized counterpart can be referenced the same way as `![my image](image.png)` on both `index.md` and `index.fr.md` when using the `suffix` docs structure.
 
 It works the same for the `folder` structure!
+
+
+## Translating admonitions
+
+This sub-option is a key/value mapping set per language and allows you to translate admonition titles which don't have an explicit title defined.
```
[the remark this answers is not in the document]
Comment (michalfapso) [reply docs/setup/translating-content.md:37]: The link makes sense. Which one would you prefer?
- mkdocs: https://www.markdownguide.org/tools/mkdocs/#using-admonitions
- python-markdown admonition extension: https://python-markdown.github.io/extensions/admonition/
- material theme: https://squidfunk.github.io/mkdocs-material/reference/admonitions

Yes, issuing the warning makes sense when admonition_translations are used without `"admonition" in config["markdown_extensions"]`. I've put the check into `plugin.py::on_page_markdown()`, not sure if that's the right place though.

**Comment by michalfapso** [comment]
Thanks for your comments, @ultrabug. I've tried to apply them in a new commit.

Do you prefer having separate commits of these subsequent changes or should I rebase and keep it in a single commit?

**Comment by ultrabug** [comment]
> Do you prefer having separate commits of these subsequent changes or should I rebase and keep it in a single commit?

Rebase please, or I'll squash & merge np

File: docs/setup/translating-content.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ultrabug) [reply docs/setup/translating-content.md:37]: - plz point to https://python-markdown.github.io/extensions/admonition/

The check should be done on config validation please

**Comment by ultrabug** [comment]
Almost there!

File: docs/setup/translating-content.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (michalfapso) [reply docs/setup/translating-content.md:37]: For the checking, in `config.py`, there are `I18nPluginLanguage` and `I18nPluginConfig`, both have `validate()`, but when I `print(self)` they don't have the `markdown_extensions` key in them.

In `plugin.py`, there's `on_config()` which gets the full config as a parameter and it includes the `markdown_extensions` so I could move the check there. 

But I assume you wanted to have this check in `config.py`. Could you please help me getting to the config's `markdown_extensions` key from there if that's possible?

[the remark this answers is not in the document]
Comment (ultrabug) [reply docs/setup/translating-content.md]: `on_config()` is fine to issue the warning, thanks!

[the remark this answers is not in the document]
Comment (michalfapso) [reply docs/setup/translating-content.md]: Thanks, I've moved it to `on_config()`. Should I also write a test for this check and warning?

[the remark this answers is not in the document]
Comment (michalfapso) [reply docs/setup/translating-content.md]: Sure :) the test is there now. I don't know how should I run the linter locally so I expect it to have some objections :)

**Comment by ultrabug** [comment]
All green, one final review and we should be good to go

**Comment by ultrabug** [comment]
Thanks and congratulations again @michalfapso

**Comment by michalfapso** [comment]
Thanks a lot, @ultrabug! It was my pleasure :)

**Pull request merged**